src/app/parcels/parcelsTable/conditionalStyling.ts

```diff
@@ -0,0 +1,68 @@
+import { DefaultTheme } from "styled-components/dist/types";
+import { ParcelsTableRow } from "./types";
+
+const getRowBoundaries = (columnValues: (string | number | null)[]): number[] => {
+    let temp = columnValues[0];
+    const boundaries: number[] = [];
+
+    columnValues.forEach((columnValue, index) => {
+        if (columnValue !== temp) {
```

Comment:
do we even need temp? can we use 

```typescript
columnValues.forEach((columnValue, index) => {
    const previousIndex = index - 1 >= 0  ? index -1 : 0;
    const previousValue = columnValues[previousIndex];
    if (columnValue !== previousValue) {
        ...
    }
etc
``` 

This avoids the use of `let` which I think is discouraged to use where possible